### PR TITLE
Fix inconsistencies in the rest api specs for *_script

### LINKED ISSUE #26923: Inconsistent API spec for `*_script`
Looking into #25737 I notice that the `*_script` rest-api-specs [1] definition of path parts as required  is inconsistent with others rest-api-specs.

For example [index](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/index.json) defines two different `paths` : `"paths": ["/{index}/{type}", "/{index}/{type}/{id}"]` and sets `index` and `type` to required, leaving `id` as an optional path part (as it is not in both `paths`).
However `get_script` although defining only a single path : `"paths": [ "/_scripts/{id}" ]`, sets `id` *and* `lang` as required.

My interpretation of the `required : true` is that the path part *must* be in all paths. 

As till now the `required` has not been enforced by the rest test runner, this has gone undetected. However for #25737 I am considering enforcing `required` both for path parts and parameters. This means I will run into more inconsistent definitions of the `required` for path parts.

My question here is: is my understanding of `required` in the rest-api-specs correct? If so, I would proceed and correct all inconsistent definitions of `required` I run into during the implementation of #25737.

[1]*_script rest-api-specs:
[delete_script](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/delete_script.json)
[get_script](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/get_script.json)
[put_script](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json)

### PR DESCRIPTION
**Pull request by olcbean**
Removing inconsistencies in the resp api specs for the *_script by setting some path parts to optional.

Fixes #26923

CC @javanna @jdconrad @Mpdreamz

### CODE REVIEW

**Review by jdconrad — APPROVED** [review]
LGTM!  Thanks for fixing this.

File: rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json [hunk 1]
```diff
@@ -11,10 +11,9 @@
           "description" : "Script ID",
           "required" : true
         },
-        "lang" : {
+        "context" : {
           "type" : "string",
-          "description" : "Script language",
-          "required" : true
+          "description" : "Script language"
```
Comment (Mpdreamz) [inline rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json]: Its not quite the `Script language` as `string`.  It's a new paramater that will cause the passed script to be compiled and evaluated in a certain `ScriptContext`. 

See: https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/script/ScriptModule.java#L41-L52

This should be an `enum` with multiple values:

`search`, `aggs`, `executable`, `aggs_executable`, `update`, `ingest`, `filter`, `similarity`, `similarity_weight` & `template`. 

Thats the current list for `master` `6.0` and `6.x`  have a slightly smaller list:

https://github.com/elastic/elasticsearch/blob/6.0/core/src/main/java/org/elasticsearch/script/ScriptModule.java#L41-L52

https://github.com/elastic/elasticsearch/blob/6.x/core/src/main/java/org/elasticsearch/script/ScriptModule.java#L41-L52

**Review by Mpdreamz — CHANGES_REQUESTED** [review]

**Comment by Mpdreamz** [comment]
Thanks for also taking `context` on board @olcbean 👍 requested some changes regarding it though.

File: rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json (hunk 1 above)
Comment (rjernst) [reply rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json]: An enum is not correct. These values are pluggable.

**Comment by olcbean** [comment]
@Mpdreamz nice catch! I was concentrated on the functionality, ignoring the descriptions ;) Can you have another look?

File: rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json (hunk 1 above)
Comment (olcbean) [reply rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json]: @Mpdreamz thanks for putting it in context!

**Review by Mpdreamz — APPROVED** [review]

File: rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json (hunk 1 above)
Comment (Mpdreamz) [reply rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json]: Thanks @rjernst, @olcbean LGTM now 👍

**Review by Mpdreamz — APPROVED** [review]

**Comment by rjernst** [comment]
@elasticmachine please test this

**Comment by rjernst** [comment]
Thanks @olcbean

**Pull request merged**